CHANGELOG.md

```diff
@@ -21,7 +21,7 @@ Canonical reference for changes, improvements, and bugfixes for Boundary.
   the ID are passed through to the type-specific subcommand. Once the ID has
   been entered, autocomplete is also supported.
 
-## 0.14.2 (Not Yet Released)
+## 0.14.2 (2023/11/2)
```

Comment:
The release notes is just missing a New & Improved section for exposing vault principals. 
Could you please add it to it. I have it in my PR: https://github.com/hashicorp/boundary/pull/3944/files. I'll close it so we could just use your PR to update the changelog

Reply:
Apologies @elimt -- I can close this in favor of yours